tools/bridge/tests/test_event_fetcher.py

```diff
@@ -0,0 +1,310 @@
+            event_name=transfer_event_name,
+            event_argument_filter=transfer_event_argument_filter,
+            event_queue=transfer_event_queue,
+            max_reorg_depth=foreign_chain_max_reorg_depth,
+        )
+
+
+def test_instantiate_event_fetcher_with_not_existing_event_name(
+    w3_foreign,
+    token_contract,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    with pytest.raises(ValueError):
+        return EventFetcher(
+            web3=w3_foreign,
+            contract_address=token_contract.address,
+            contract_abi=MINIMAL_ERC20_TOKEN_ABI,
+            event_name="NonExistingEvent",
+            event_argument_filter=transfer_event_argument_filter,
+            event_queue=transfer_event_queue,
+            max_reorg_depth=foreign_chain_max_reorg_depth,
+        )
+
+
+def test_instantiate_event_fetcher_with_negative_event_fetch_limit(
+    w3_foreign,
+    token_contract,
+    transfer_event_name,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    with pytest.raises(ValueError):
+        return EventFetcher(
+            web3=w3_foreign,
+            contract_address=token_contract.address,
+            contract_abi=MINIMAL_ERC20_TOKEN_ABI,
+            event_name=transfer_event_name,
+            event_argument_filter=transfer_event_argument_filter,
+            event_fetch_limit=-1,
+            event_queue=transfer_event_queue,
+            max_reorg_depth=foreign_chain_max_reorg_depth,
+        )
+
+
+def test_instantiate_event_fetcher_with_negative_max_reorg_depth(
+    w3_foreign,
+    token_contract,
+    transfer_event_name,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    foreign_chain_max_reorg_depth,
+):
+    with pytest.raises(ValueError):
+        return EventFetcher(
+            web3=w3_foreign,
+            contract_address=token_contract.address,
+            contract_abi=MINIMAL_ERC20_TOKEN_ABI,
+            event_name=transfer_event_name,
+            event_argument_filter=transfer_event_argument_filter,
+            event_queue=transfer_event_queue,
+            max_reorg_depth=-1,
+        )
+
+
+def test_fetch_events_in_range(
+    transfer_event_fetcher,
+    w3_foreign,
+    transfer_event_name,
+    transfer_event_argument_filter,
+    transfer_event_queue,
+    transfer_tokens_to_foreign_bridge,
+):
+    assert transfer_event_queue.empty()
+
+    transfer_tokens_to_foreign_bridge()
+    transfer_event_fetcher.fetch_events_in_range(0, w3_foreign.eth.blockNumber)
+
+    assert transfer_event_queue.qsize() == 1
+
+    event = transfer_event_queue.get()
+
+    assert event["event"] == transfer_event_name
+
+    for argument_name, argument_value in transfer_event_argument_filter.items():
+        print(f"Name: {argument_name} Value: {argument_value}")
+        event.args[argument_name] == argument_value
```

Comment:
Is there an assert missing? At the moment this is a NOP?

Reply:
Yes. Inserted it while removing the obsolete print statements. Haven't observe ed it while scrolling to my changes.

Reply:
:heavy_check_mark: